data/pools/transactionPool.go

```diff
@@ -582,16 +593,23 @@ func (pool *TransactionPool) addToPendingBlockEvaluatorOnce(txgroup []transactio
 					stats.StopReason = telemetryspec.AssembleBlockFull
 				} else {
 					stats.StopReason = telemetryspec.AssembleBlockTimeout
+					// if the block is not full, it means that the above transaction made it to the block, so we want to add it here.
+					stats.ProcessingTime.AddTransaction(transactionGroupDuration)
 				}
-				pool.assemblyResults.stats = *stats
 
+				blockGenerationStarts := time.Now()
 				lvb, gerr := pool.pendingBlockEvaluator.GenerateBlock()
 				if gerr != nil {
 					pool.assemblyResults.err = fmt.Errorf("could not generate block for %d: %v", pool.assemblyResults.roundStartedEvaluating, gerr)
 				} else {
 					pool.assemblyResults.blk = lvb
 				}
+				stats.BlockGenerationDuration = uint64(time.Now().Sub(blockGenerationStarts))
+				pool.assemblyResults.stats = *stats
 				pool.assemblyCond.Broadcast()
+			} else {
+				// add the transaction time only if we didn't ended up finishing the block.
+				stats.ProcessingTime.AddTransaction(transactionGroupDuration)
```

Comment:
Why stats in not set in this branch?
`pool.assemblyResults.stats = *stats`

Reply:
because there is no need to do that yet.. it will get updated when the block get finalized.